protocol/xpush/xpush.go

```diff
@@ -261,6 +261,7 @@ func (s *socket) Close() error {
 		return protocol.ErrClosed
 	}
 	s.closed = true
+	s.cv.Signal()
```

Comment:
This should be s.cv.Broadcast() (because what if multiple threads are waiting to send?)